app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,8 @@
+class AddressesController < APIController
+  before_action :doorkeeper_authorize!, only: [:index]
+
+  def index
+    authorize Address
+    render json: Address.all
```

Comment:
A `WalkingList` I think would indeed be a list of addresses, but I think it would be more like `WalkingList.for_user(user)`, which `has_many addresses`.

The difficulty again lies in the order of operations, though. How do we isolate addresses that a campaign needs canvassed and that someone on their campaign is not already canvassing, AND that are close to the person canvassing, AND are within the boundaries of that campaign? I don't think we have a good answer for this right now.

We can address in a separate issue, if you'd like.